derive/src/derive.rs

```diff
@@ -74,49 +74,47 @@ pub fn tovalue_derive(mut s: synstructure::Structure) -> proc_macro::TokenStream
             panic!("ocaml cannot derive unboxed or float arrays for enums")
         }
         if arity == 0 {
-            let init = quote!(value = ocaml::Value::int(#tag as ocaml::Int));
+            let init = quote!(value = unsafe { ocaml::Value::int(#tag as ocaml::Int)});
             variant.fold(init, |_, _| quote!())
         } else if attrs.floats {
             let mut idx = 0usize;
             let init = quote!(
-                value = ocaml::Value::alloc(#arity, ocaml::Tag::DOUBLE_ARRAY);
+                value = unsafe { ocaml::Value::alloc(gc, #arity, ocaml::Tag::DOUBLE_ARRAY) };
             );
             variant.fold(init, |acc, b| {
                 let i = idx;
                 idx += 1;
-                quote!(#acc; ocaml::array::set_double(value, #i, *#b as f64).unwrap();)
+                quote!(#acc; ocaml::array::set_double(gc, value, #i, *#b as f64).unwrap();)
             })
         } else if attrs.unboxed {
             if variant.bindings().len() > 1 {
                 panic!("ocaml cannot unboxed record with multiple fields")
             }
-            variant.each(|field| quote!(#field.to_value()))
+            variant.each(|field| quote!(unsafe {#field.into_value(gc) }))
```

Comment:
@zshipko is this call really `unsafe`? the `fn into_value(...)` declarations that show up on this diff don't have the `unsafe` qualifier.

Reply:
Thanks, that doesn't need to be in an unsafe block